### PR TITLE
Fix Jetpack sharing options checkbox when API returns object

### PR DESCRIPTION
**Pull request by seear**
On jetpack sites with the _WooCommerce_ plugin active, the site settings endpoint can return an object instead of an array for the `sharing_show` field:

```
            [sharing_show] => stdClass Object
                (
                    [0] => page
                    [1] => index
                    [2] => attachment
                    [3] => post
                    [5] => product
                )
```

This object is passed through to the multi-checkbox component at https://wordpress.com/sharing/buttons/{site}, which only works with arrays, making it impossible to change the selections.

checkboxes

This PR ensures that object returns are converted to arrays.

## Testing
* Connect a dotorg site with Jetpack and WooCommerce plugins active
* Go to https://wordpress.com/sharing/buttons/{site}
* Select every checkbox in _Show sharing buttons on_
* Save changes
* Attempt to select another checkbox
* There should be no js errors in the console, and the box should be selectable

### CODE REVIEW

File: client/state/site-settings/test/utils.js [hunk 1]
```diff
@@ -27,9 +22,29 @@ describe( 'utils', () => {
...
+			expect( normalizeSettings( settings ) ).toEqual( {
 				default_category: 10,
 			} );
 		} );
+
+		test( 'should not touch sharing_show array', () => {
+			const settings = {
+				sharing_show: [ 'page', 'index', 'post' ],
+			};
+			expect( normalizeSettings( settings ) ).toEqual( settings );
```
[the remark this answers is not in the document]
Comment (seear) [reply client/state/site-settings/test/utils.js]: > We want to literally return the same array

That would be an implementation detail, though? :)

[the remark this answers is not in the document]
Comment (sirreal) [reply client/state/site-settings/test/utils.js]: Can be relevant in Redux context where referential equality can save or provoke rerenders, but I'm not sure if that's involved here.

[the remark this answers is not in the document]
Comment (seear) [reply client/state/site-settings/test/utils.js]: `toBe` fails, as does `expect( normalizeSettings( settings ).sharing_show ).toBe( settings.sharing_show );`

Why?

File: client/state/site-settings/utils.js [hunk 2]
```diff
@@ -12,6 +18,13 @@ export function normalizeSettings( settings ) {
 			case 'default_category':
 				memo[ key ] = parseInt( settings[ key ] );
 				break;
+			case 'sharing_show':
+				if ( typeof settings[ key ] === 'object' ) {
```
Comment (sirreal) [inline client/state/site-settings/utils.js]: This check is going to convert arrays.
```js
typeof []
// object
```

[`isPlainObject`](https://lodash.com/docs/4.17.10#isPlainObject) might be a good choice

File: client/state/site-settings/test/utils.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (sirreal) [reply client/state/site-settings/test/utils.js]: We're converting Arrays as well. Nice referential equality test 😁

**Pull request merged**